src/stan_math_backend/Lower_functions.ml

```diff
@@ -267,20 +268,34 @@ let collect_functors_functions (p : Program.Numbered.t) : defn list =
       | {fdname; fdbody= Some _; _} -> Some fdname
       | _ -> None )
     |> String.Set.of_list in
+  (* overloaded functions generate only one functor struct per name *)
+  let structs = String.Table.create () in
+  let is_overload (d : _ Program.fun_def) arg_types =
+    List.equal UnsizedType.equal
```

Comment:
If this is testing for equality of the argument lists then isn't it testing that the function is actually the same, rather than testing that it is an overload?

Reply:
Yes. This is definitely a case where the function name is just a random word that was on my mind while writing the code. `is_correct_overload` would have been more accurate.